### ISSUE TITLE
possible bugs in "conda env" and "conda create"

### ISSUE THREAD

**Issue by megies** [conda/conda#2826]
I've come across this problem recently (might be connected to/triggered by ContinuumIO/anaconda-issues#838?):
```
~$ $CONDAMAINBIN/conda create -y -n $CONDATMPNAME --clone py3-docs-master
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ....
src_prefix: '/home/obspy/anaconda3/envs/py3-docs-master'
dst_prefix: '/home/obspy/anaconda3/envs/tmp_build_docs-pr'
Packages: 163
Files: 532
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/home/obspy/anaconda3/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/cli/main.py", line 139, in main
    args_func(args, p)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/cli/main.py", line 146, in args_func
    args.func(args, p)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/cli/main_create.py", line 49, in execute
    install.install(args, parser, 'create')
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/cli/install.py", line 247, in install
    clone(args.clone, prefix, json=args.json, quiet=args.quiet, index=index)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/cli/install.py", line 84, in clone
    quiet=quiet, index=index)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/misc.py", line 246, in clone_env
    sorted_dists = r.dependency_sort(dists)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/resolve.py", line 729, in dependency_sort
    depends = lookup(value)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/resolve.py", line 724, in lookup
    return set(ms.name for ms in self.ms_depends(value + '.tar.bz2'))
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/resolve.py", line 573, in ms_depends
    deps = [MatchSpec(d) for d in self.index[fn].get('depends', [])]
KeyError: 'proj.4-4.9.1-0.tar.bz2'

~$ $CONDAMAINBIN/conda info
Using Anaconda Cloud api site https://api.anaconda.org
Current conda install:

             platform : linux-32
        conda version : 4.0.5
  conda-build version : 1.19.0
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : /home/obspy/anaconda3  (writable)
  default environment : /home/obspy/anaconda3
     envs directories : /home/obspy/anaconda3/envs
        package cache : /home/obspy/anaconda3/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-32/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-32/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

```

Before filing an issue I tried updating `conda`, but this triggers a different error now:

```
~$ $CONDAMAINBIN/conda env remove -y -n $CONDATMPNAME
Fetching package metadata .......
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/home/obspy/anaconda3/bin/conda-env", line 6, in <module>
    sys.exit(conda_env.cli.main.main())
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda_env/cli/main.py", line 68, in main
    return args_func(args, parser)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda/cli/main.py", line 127, in args_func
    args.func(args, p)
  File "/home/obspy/anaconda3/lib/python3.5/site-packages/conda_env/cli/main_remove.py", line 73, in execute
    plan.PREFIX: prefix,
AttributeError: module 'conda.plan' has no attribute 'PREFIX'

~$ $CONDAMAINBIN/conda info
Current conda install:

             platform : linux-32
        conda version : 4.1.2
  conda-build version : 1.19.0
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : /home/obspy/anaconda3  (writable)
  default environment : /home/obspy/anaconda3
     envs directories : /home/obspy/anaconda3/envs
        package cache : /home/obspy/anaconda3/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-32/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-32/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

```

**Comment by mcg1969** [comment]
The second bug is known and fixed in `conda-env 2.5.1`. I think you're right about the diagnosis of the first problem. Did the fix offered in the Anaconda issue help for you at all?

**Comment by megies** [comment]
> The second bug is known and fixed in conda-env 2.5.1.

Oh.. sorry for not checking. Shouldn't `conda-env` also get updated when I do a `conda update conda` though..?

In any case, I'm not running into any of the above problems with the updated `conda-env` anymore. Thanks a lot for pointing me in the right direction @mcg1969!

**Comment by mcg1969** [comment]
> Oh.. sorry for not checking. Shouldn't conda-env also get updated when I do a conda update conda though..?

Yes and no. :-) Previous versions of `conda` had a strategy of performing the minimal set of updates to bring an environment up to date. So if upgrading only to conda 2.5.0 was sufficient, that's what it would do. 

We've now adjusted the algorithm so that, instead of minimizing the *total version bumps*, it just minimizes the *number of packages changed*---but updates those packages to the maximum. So *that* would indeed pick up the latest conda-env.

**Comment by megies** [comment]
FYI, above problems are fixed for me now with conda 4.1.3 and conda-env 2.5.1. Thanks! Keep it up with conda, great work! :sparkles:
